src/components/PostPageDetail/index.js

```diff
@@ -576,6 +579,11 @@ const PostPageDetailIdComponent = (props) => {
             isBackButton={true}
             source={SOURCE_PDP}
             height={getHeightHeader()}
+            isFollow={item?.is_following_target || parentData?.is_following_target}
+            onPressFollUnFoll={() => {
+              followUnfollow(item);
+              setItem({...item, is_following_target: !item?.is_following_target});
+            }}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is a potential race condition here. The `followUnfollow` function is likely asynchronous, and it might not complete before the state update on line 585. This could lead to inconsistent states between your local state and the server.

To fix this, you should wait for the `followUnfollow` function to complete before updating the state. You can do this by making the `onPressFollUnFoll` function async and then await the `followUnfollow` call:

```diff
- onPressFollUnFoll={() => {
-   followUnfollow(item);
-   setItem({...item, is_following_target: !item?.is_following_target});
- }}
+ onPressFollUnFoll={async () => {
+   await followUnfollow(item);
+   setItem(prevItem => ({...prevItem, is_following_target: !prevItem?.is_following_target}));
+ }}
```
Note that I also changed the `setItem` call to use a function that takes the previous state as an argument. This ensures that we're always working with the most recent state when we toggle the `is_following_target` property.


<!-- This is an auto-generated comment by OSS CodeRabbit -->